src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -367,3 +369,62 @@ def count_web_vitals_snql_factory(aggregate_filter, org_id, measurement_rating,
         ],
         alias=alias,
     )
+
+
+def count_transaction_with_condition_snql_factory(aggregate_filter, org_id, condition, alias=None):
+    is_unparameterized = "is_unparameterized"
+    is_null = "is_null"
+    has_value = "has_value"
+
+    def _generate_transaction_tag_value_filter(operation, inner_condition):
```

Comment:
I think we could replace it with `transaction_name_condition` because if we mix `condition` and `type` it could look a bit difficult to understand it in a few weeks.

The only issue I find with the use of `condition` is that it might be a bit misleading given the fact that a condition is then mapped onto a string with a different values. I wanted to use enums on the `MetricField` side but they are not supported by the type definition and I didn't want to make such a change for a small detail.

Reply:
I guess you could name it `transaction_name_identifier` ?

Reply:
> I wanted to use enums on the MetricField side but they are not supported by the type definition and I didn't want to make such a change for a small detail.

I like this but I agree let's do it later on

Reply:
> > I wanted to use enums on the MetricField side but they are not supported by the type definition and I didn't want to make such a change for a small detail.
> 
> I like this but I agree let's do it later on

I will add it to the document of possible improvements.